articles/lipa-na-mpesa-online/index.md

```diff
@@ -0,0 +1,217 @@
+                message:error
+            });
+
+            //else true
+            return res.send({
+                success:true,
+                message:body
+            })
+        })
+}
+```
+
+The implementation follows the following steps:
+- Access the token which we stored in the request object.
+- Append Bearer in front of the string to create the authorization string.
+- Get the current timestamp.
+- Get the lipa na Mpesa URL which is stored as an environmental variable.
+- Get your business short code and `passkey`. To get them, proceed to  your [test credentials](https://developer.safaricom.co.ke/test_credentials). Then copy and paste the `Lipa Na Mpesa Online Shortcode` as the business short code to your `.env` file and `Lipa Na Mpesa Online Passkey` as the passkey.
+- Get the password which is generated from a buffer composed of business short code, passkey, and timestamp. The buffer is then encoded to a `base64` string.
+- Set the transaction type to CustomerPayBillOnline which is the only one supported. Don't change it.
+- Set the amount to be paid.
+- Set the PartyA which is the phone number sending the funds. The phone number should use the `2547xxxxxxxx` format.
+- Set the PartyB which is the business short code.
+- Set the Phone number which is the phone number sending the funds. Use the format as explained in PartyA.
+- Set the callback Url. The callback Url is where the response shall be processed. For this particular case we shall use [ngrok](https://ngrok.com/). Make sure your server is running with the command:
+
+```bash
+npm run dev
+```
+
+Then open another tab and run the following command:
+
+```bash
+npm run ngrok
+```
+
+This shall forward you to a particular link such as `http://78066c1c2d6b.ngrok.io`. Copy that link and paste in the callBackUrl variable. 
+
+**Note that the link is always different each time you stop and start the server again.**
+
+- Set the account reference which can be any string you set. It references where the money is being paid to.
+- Set the transaction description which describes what the transaction is for.
+- Use `request` to send the data to the Daraja API. 
+- Set the method, headers, and `JSON` body to send.
+- Receive the result from the API in form of `error`,`response`,  and `body`. If there is an error, send back the error, else send the body object.
+
+#### Lipa na Mpesa online callback.
+When we send a request, the response from the API shall be sent through the callback. In case we want to update some records, we update them here based on the response sent back. We implement a simple callback as a method on the class.
+
+```javascript
+lipaNaMpesaOnlineCallback(req,res){
+        
+    let message = req.Body.stkCallback.ResultDesc;
+
+    //based on the message you can update some records.
+
+    return res.send({
+        success:true,
+        message
+    });
+        
+};
+```
+
+- Destructure the message sent from the request object.
```

Comment:
You are not exactly destructuring the object rather than accessing the values in the object. Change the terminology as it might confuse readers

Reply:
fixed it.